### ISSUE TITLE
The build name is not set on a certain build failure

### ISSUE THREAD

**Issue by jansvajcr** [jenkinsci/build-name-setter-plugin#57]
These are my recent builds:

![Výstřižek](https://user-images.githubusercontent.com/5638104/74085005-f547b300-4a74-11ea-8019-6302c0dcca43.JPG)

The last one failed because of some git fetch error. Logically thinking, the build name could have been set because I only wanted it to be a timestamp value (`yyyy-MM-dd`) which was known by the time of build failure.

Is it technically possible to make this working?

**Comment by damianszczepanik** [comment]
I think this kind of error (I was able to reproduce it) is happening before the build name is set so that might be hard. Probably major refactor would be needed.

**Comment by paulvi** [comment]
@jansvajcr Was the pre-build action configured (not post action)?
Because for post-build action may be skipped in build has failed.

**Comment by jansvajcr** [comment]
> @jansvajcr Was the pre-build action configured (not post action)? Because for post-build action may be skipped if build has failed.

No, I set it via the _Set Build Name_ option in the _Build Environment_ job configuration section.
